tests/utils/fixture.ts

```diff
@@ -405,8 +411,20 @@ export async function invokeFunction(
       },
       lambdaFunc: { handler },
       timeoutMs: 4_000,
+      onInvocationEnd: (error) => {
+        // lambda-local resolve promise return from execute when response is closed
+        // but we should wait for tracked background work to finish
+        // before resolving the promise to allow background work to finish
+        if (error) {
+          rejectInvocation(error)
+        } else {
+          resolveInvocation()
+        }
+      },
     })) as LambdaResponse
 
+    await invocationPromise
```

Comment:
possibly instead of awaiting here we should return this promise, but before https://github.com/netlify/serverless-functions-api/pull/358 landed we were basically waiting for stream to close here and we were blocking stream closing to ensure background work